spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -0,0 +1,653 @@
+        fname_out = arguments.o
+    else:
+        path, file_name, ext = extract_fname(get_absolute_path(arguments.i))
+        fname_out = os.path.join(path, file_name + '_compression_metrics' + '.csv')
+
+    # Step 1. Load subject input files and get csv files (label image, segmentation)
+    # -----------------------------------------------------------
+    img_seg = Image(fname_segmentation).change_orientation('RPI')
+    img_labels = Image(fname_labels).change_orientation('RPI')
+    img_vertfile = Image(fname_vertfile).change_orientation('RPI')
+    # Check if segmentation, compression labels, and vertebral label files have same dimensions
+    if not img_seg.data.shape == img_labels.data.shape == img_vertfile.data.shape:
+        raise ValueError(f"Shape mismatch between compression labels [{img_labels.data.shape}], vertebral labels [{img_vertfile.data.shape}]"
+                         f" and segmentation [{img_seg.data.shape}]). "
+                         f"Please verify that your compression labels and vertebral labels were done in the same space as your input segmentation.")
+    # Call sct_process_segmentation to get morphometrics perslice in native space
+    path, file_name, ext = extract_fname(get_absolute_path(arguments.i))
+    fname_metrics = os.path.join(path, file_name + '_metrics' + '.csv')
+    sct_process_segmentation.main(argv=['-i', fname_segmentation, '-vertfile', fname_vertfile, '-perslice', '1', '-o', fname_metrics])
+    metric = 'MEAN(' + arguments.metric + ')'  # Adjust for csv file columns name
+    # Fetch metrics of subject
+    df_metrics = pd.read_csv(fname_metrics).astype({metric: float})
+    # Get vertebral level corresponding to the slice with the compression
+    slice_thickness = get_slice_thickness(img_labels)
+    slice_compressed = get_compressed_slice(img_labels, verbose)
+    compressed_levels_dict = get_verterbral_level_from_slice(slice_compressed, df_metrics)
+    # Initialize variables if normalization
+    if arguments.normalize:
+        fname_metrics_PAM50 = os.path.join(path, file_name + '_metrics_PAM50' + '.csv')
+        # Call sct_process_segmentation to get morphometrics perslice in PAM50 space
+        sct_process_segmentation.main(argv=['-i', fname_segmentation, '-vertfile', fname_vertfile, '-normalize-PAM50', '1',
+                                      '-perslice', '1', '-o', fname_metrics_PAM50])
+        sex = arguments.sex
+        age = arguments.age
+
+        # Step 2. Load reference input files (PAM50, healthy controls)
+        # ------------------------------------------------------------
+        # Select healthy controls based on sex and/or age range
+        path_ref = os.path.join(__data_dir__, 'PAM50_normalized_metrics')
+        if sex or age:
+            if not os.path.isfile(get_absolute_path(os.path.join(path_ref, arguments.file_participants))):
+                raise FileNotFoundError('participants.tsv file must exists to select sex or age.')
```

Comment:
I think this input check should also be moved to the "argument validation section"